crypto/src/key/mod.rs

```diff
@@ -48,7 +48,7 @@ pub struct PrivateKey {
     key: PrivateKeyHolder,
 }
 
-#[derive(Debug, PartialOrd, Ord, PartialEq, Eq, Clone, Decode, Encode)]
+#[derive(Debug, PartialOrd, Ord, PartialEq, Eq, Hash, Clone, Decode, Encode)]
```

Comment:
We don't use `Hash` at all to avoid confusion with proper cryptography hashes. In practice, there's barely any need for this. If you need a HashSet or HashMap, please use BTreeSet or BTreeMap instead.